### PR TITLE
Remove use of python_cmake_module

### PR DESCRIPTION
**Pull request by clalancette**
Also add in hints so that we find the unversioned python3 before finding versioned ones.

This must be merged before https://github.com/ros2/ros2/pull/1524 ; see that pull request for more information about this change.

### CODE REVIEW

**Review by fujitatomoya — APPROVED** [review]
lgtm with green CI

CC: @MichaelOrlov

**Review by MichaelOrlov — APPROVED** [review]
LGTM, if CI will pass green.

**Comment by clalancette** [comment]
I *think* in order for this to work on Windows Debug, we are also going to need https://github.com/ros2/ci/pull/755 first.  So let's hold off on merging this for now.

**Comment by clalancette** [comment]
CI for this (along with all of the rest of the changes needed) is in https://github.com/ros2/ros2/pull/1524#issuecomment-2365359588

**Pull request merged**
